rfcs/9999-new-merkle-storage.md

```diff
@@ -0,0 +1,140 @@
+## Summary
+[summary]: #summary
+
+I propose creating a new, generic, system for storing arbitrary Merkleized data.
+It would unify and replace the existing system of mask, database, null, and
+sparse ledgers, as well as the Merkle lists we use.
+
+## Motivation
+
+[motivation]: #motivation
+
+The current system is complex, error prone and hard to debug. It also doesn't
+exploit redundancy if you don't explicitly use masks. For example we expect the
+epoch and snarked ledgers to be very similar, there is probably a power law
+distribution on how active accounts are, the vast majority will not have changed
+at all since the epoch snapshot was taken. But in the current system it'd be
+very difficult to exploit that.
+
```

Comment:
The point is, data is stored exactly once, and never has to be moved between data structures.